ste/xfer-anyToRemote-file.go

```diff
@@ -363,12 +365,15 @@ func epilogueWithCleanupSendToRemote(jptm IJobPartTransferMgr, s sender, sip ISo
 		_, isS2SCopier := s.(s2sCopier)
 		destLength, err := s.GetDestinationLength()
 
-		if err != nil {
+		if stgErr, stgErrOk := err.(azfile.StorageError); stgErrOk && stgErr.Response() != nil {
```

Comment:
There are cases where err will have a value, but that value will not be a StorageError. E.g. it might be a network error.

As far as i can tell this current code deosn't really handle those errors.  It just skips the length check if they happen, but it does say anything.

I'd feel more comfortable with this if we hav some kind of Boolean, say shouldCheckLength := true.  And then immediately after s.GetDestinationLength we set it to false if the error is a StorageError with http.StatusForbidden. If shouldCheckLength is true, then we'd do the existing logic.
I.e. 
```
shouldCheckLength := true
destlength, err := s.GetDestinatinoLength
if stgErr, stgErrOk := err.(pipeline.Response); stgErrOk && stgErr.Response() != nil && stgErr.Response().StatusCode != http.StatusForbidden {
  shouldCheckLength = false
}
if shouldCheckLength { 
     if err != nil { 
        existing error code block, from before this PR..
     } else {
        do the length check 
    }
}
```